flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/catalog/PostgresCatalog.java

```diff
@@ -216,6 +234,49 @@ public CatalogBaseTable getTable(ObjectPath tablePath) throws TableNotExistExcep
 		}
 	}
 
+	private Map<String, List<String>>  getUniqueConstraints(DatabaseMetaData metaData, PostgresTablePath pgPath, Map.Entry<String, List<String>> pk) throws SQLException {
```

Comment:
Maybe it could worth to move getUniqueConstraints() and getPrimaryKey() into AbstractJDBCCatalog since they are of general validity